packages/query/src/leather-query-provider.tsx

```diff
@@ -0,0 +1,68 @@
+import { ReactNode, createContext, useContext, useMemo } from 'react';
+
+import { NetworkConfiguration, NetworkModes } from '@leather-wallet/constants';
+import { ChainID } from '@stacks/common';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+
+const LeatherNetworkContext = createContext<NetworkConfiguration | null>(null);
+
+const LeatherEnvironmentContext = createContext<string | null>(null);
+
+export function useLeatherEnv() {
+  const leatherEnv = useContext(LeatherEnvironmentContext);
+
+  if (!leatherEnv) {
+    throw new Error('No LeatherEnvironment set, use LeatherQueryProvider to set one');
+  }
+
+  return leatherEnv;
+}
+
+export function useIsLeatherTestingEnv() {
+  const leatherEnv = useContext(LeatherEnvironmentContext);
+
+  if (!leatherEnv) {
+    throw new Error('No LeatherEnvironment set, use LeatherQueryProvider to set one');
+  }
+
+  return leatherEnv === 'testing';
+}
+
+export function useLeatherNetwork() {
+  const leatherNetwork = useContext(LeatherNetworkContext);
+
+  if (!leatherNetwork) {
+    throw new Error('No LeatherNetwork set, use LeatherQueryProvider to set one');
+  }
+
+  return leatherNetwork;
+}
+export function useCurrentNetworkState() {
+  const currentNetwork = useLeatherNetwork();
+
+  return useMemo(() => {
+    const isTestnet = currentNetwork.chain.stacks.chainId === ChainID.Testnet;
+    const mode = (isTestnet ? 'testnet' : 'mainnet') as NetworkModes;
+    return { ...currentNetwork, isTestnet, mode };
+  }, [currentNetwork]);
+}
```

Comment:
What if in the wallet, we need to query mainnet and testnet at the same time? I think there are one or two places where we need to do this. Maybe we should have different hooks, or the hooks accept a param? 🤔